charts/redpanda/templates/_helpers.tpl

```diff
@@ -582,6 +585,97 @@ return a warning if the chart is configured with insufficient CPU
   {{- toJson $brokers -}}
 {{- end -}}
 
+{{- define "kafka-brokers-sasl-enabled" -}}
+  {{- $root := . -}}
+  {{- $kafkaService := .Values.listeners.kafka }}
+  {{- $auditLogging := .Values.auditLogging }}
+  {{- $brokers := list -}}
+  {{- $broker_tls := dict -}}
+  {{- $result := dict -}}
+  {{- $tlsEnabled := .Values.tls.enabled -}}
+  {{- $tlsCerts := .Values.tls.certs -}}
+  {{- $trustStoreFile := "" -}}
+  {{- $requireClientAuth := dig "tls" "requireClientAuth" false $kafkaService -}}
+  {{- if and ( eq "internal" $auditLogging.listener ) ( eq (default "sasl" $kafkaService.authenticationMethod) "sasl" ) -}}
+    {{- range $address := (include "seed-server-list" $root | mustFromJson) }}
+      {{- $brokerItem := ( dict
+        "address" $address
+        "port" $kafkaService.port
+        )
+      -}}
+      {{- $brokers = append $brokers $brokerItem -}}
+    {{- end -}}
+    {{- if $brokers -}}
+      {{- $result = set $result "brokers" $brokers -}}
+    {{- end -}}
+    {{- if dig "tls" "enabled" $tlsEnabled $kafkaService -}}
+      {{- $cert := get $tlsCerts $kafkaService.tls.cert -}}
+      {{- if empty $cert -}}
+        {{- fail (printf "Certificate, '%s', used but not defined") -}}
+      {{- end -}}
+      {{- if $cert.caEnabled -}}
+        {{- $trustStoreFile =  ( printf "/etc/tls/certs/%s/ca.crt" $kafkaService.tls.cert ) -}}
+      {{- else -}}
+        {{- $trustStoreFile = "/etc/ssl/certs/ca-certificates.crt" -}}
+      {{- end -}}
+      {{- $broker_tls = ( dict
+        "enabled" true
+        "cert_file" ( printf "/etc/tls/certs/%s/tls.crt" $kafkaService.tls.cert )
+        "key_file" ( printf "/etc/tls/certs/%s/tls.key" $kafkaService.tls.cert )
+        "require_client_auth" $requireClientAuth
+        )
+      -}}
+      {{- if $trustStoreFile -}}
+        {{- $broker_tls = set $broker_tls "truststore_file" $trustStoreFile -}}
+      {{- end -}}
+      {{- if $broker_tls -}}
+        {{- $result = set $result "broker_tls" $broker_tls -}}
+      {{- end -}}
+    {{- end -}}
+  {{- else -}}
+    {{- range $name, $listener := $kafkaService.external -}}
+      {{- if and ( and ( and $listener.port $name (dig "enabled" true $listener)) ( eq (default "sasl" $listener.authenticationMethod) "sasl" )) ( eq $name $auditLogging.listener ) -}}
+        {{- range $address := (include "seed-server-list" $root | mustFromJson) }}
```

Comment:
I know other clients configuration (schema_registry and pandaproxy) have seed server list, but we excluded seed server list from statefulset annotation config-map-hash. I don't believe using chart-testing we could verify that downscaling is not triggering full rollout.

Reply:
https://github.com/redpanda-data/helm-charts/pull/692

and

https://github.com/redpanda-data/helm-charts/issues/690

Reply:
OH man thank you, i completely forgot this.

Reply:
This seems to exist in the configmap and may also be a bigger issue: 

for example:
```
  schema_registry_client:
    brokers:
      {{- range (include "seed-server-list" $root | mustFromJson) }}
    - address: {{ . }}
      port: {{  $kafkaService.port }}
      {{- end }}
```

which occurs in lines
https://github.com/redpanda-data/helm-charts/blob/5d9a78033a29f6d36bf44877574151b523f39b10/charts/redpanda/templates/_configmap.tpl#L326
https://github.com/redpanda-data/helm-charts/blob/5d9a78033a29f6d36bf44877574151b523f39b10/charts/redpanda/templates/_configmap.tpl#L425

Not sure why we did not capture this before. I will check the hash but I think i would like to fix this in another PR if you do not mind.

Reply:
I'm not sure if this is correct to introduce something that shouldn't be there, to then work on proper fix.

Never the less go for another PR.

Reply:
My point is that it already is there, did you not see the lines?

Reply:
> My point is that it already is there, did you not see the lines?

I know that schema registry and pandaproxy client is using the seed server. That was my first observation in the first comment 

https://github.com/redpanda-data/helm-charts/pull/934#discussion_r1431995422

From my point of view if we introduce third place where seed server list is part of the checksum, then we will have more work to exclude 3 places not 2. I'm aware of the problem existence even if you do the work to exclude audit_log_client, that's I'm ok with pushing it to the next PR, but not closing the whole effort.

Reply:
Fixed it, now we have the right configmaps and 
```
helm template redpanda charts/redpanda/ -n redpanda  --set image.repository=docker.redpanda.com/redpandadata/redpanda-unstable --set image.tag=v23.3.1-rc2 --set auditLogging.enabled=true --set auditLogging.listener=internal --set auth.sasl.enabled=true --set auditLogging.clientMaxBufferSize=16777215 --set statefulset.replicas=3 | grep config.redpanda.com/checksum
        config.redpanda.com/checksum: 8a0c69ef04db448f5b9136443b7d933b63a22a36fc87f2842343f86bfd6cb493
```

```
helm template redpanda charts/redpanda/ -n redpanda  --set image.repository=docker.redpanda.com/redpandadata/redpanda-unstable --set image.tag=v23.3.1-rc2 --set auditLogging.enabled=true --set auditLogging.listener=internal --set auth.sasl.enabled=true --set auditLogging.clientMaxBufferSize=16777215 --set statefulset.replicas=6 | grep config.redpanda.com/checksum
        config.redpanda.com/checksum: 8a0c69ef04db448f5b9136443b7d933b63a22a36fc87f2842343f86bfd6cb493
```

Reply:
Please see my suggestion https://github.com/redpanda-data/helm-charts/pull/934#discussion_r1432908552